### PR TITLE
Change placeholder text for login redirect

### PR DESCRIPTION
**Pull request by drmenzelit**
Pull Request for Issue #32561 comment https://github.com/joomla/joomla-cms/issues/32561#issuecomment-790059986 .

### Summary of Changes
~~Changed text in `/administrator/language/en-GB/com_menus.ini` from "Home" to "Select a menu item"~~
Changes in com_menus.ini reverted, because that was wrong.
Changed the value of the fields to "Select menu" (existing string `JOPTION_SELECT_MENU`).

As stated in the comment mentioned above the text "Home" in the redirect field can be misunderstood. The field is disabled, but it seems as the "Home" menu item is already selected as redirect page.
Added descriptions to the fields to explain the behaviour of login and logout redirect.

### Testing Instructions
Create a login menu, set "Choose Login Redirect Type" to "Menu item". 
Create a login modul.

### Actual result BEFORE applying this Pull Request
Login menu
Screenshot_2021-03-08 Menus Edit Item - Joomla 4 Beta - Administration

Login module
Screenshot_2021-03-08 Modules Login - Joomla 4 Beta - Administration

### Expected result AFTER applying this Pull Request
Login menu
Screenshot_2021-03-08 Menus Edit Item - Joomla 4 Beta - Administration3
Screenshot_2021-03-08 Menus Edit Item - Joomla 4 Beta - Administration-Logout

Login module
Screenshot_2021-03-08 Modules Login - Joomla 4 Beta - Administration3

### CODE REVIEW

**Comment by chmst** [comment]
Much better! But wouldn't it be better to use anoter language key? 
JDEAULT is one of the common language keys as JYES, JNO, JCATEGORY which have the same translation over the whole system. 
In all other language files JDEFAULT is translated wit "Default".

**Comment by brianteeman** [comment]
Does it even need a hint?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32613">issues.joomla.org/tracker/joomla-cms/32613</a>.</sub>

**Comment by drmenzelit** [comment]
"Home" as placeholder is wrong, "Default" as placeholder (if we don't override JDEFAULT) is also not self explaining. So we need something better. Which way is better: override the value of JDEFAULT or use another language string?

**Comment by chmst** [comment]
@brianteeman the hints are very old and inherited fom J3. There the hint is "default", but has added  a description what "default" means.  
grafik

Imho "the same page" is not user profile when I am on the login page - but this is not in scope here. I also always was surprised in J3 installations when the profile appeared after login. 

This PR makes it better for now and blocks nothing. 

Further improvements can be
- Remove the hint (in all xml files)
- Change the default behaviour as the j3 description falsely announces and remain on the login page 
- Change the default behaviour and set "home" as default as the wrong hint falsely promised

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 17625bfe2c55013f61a0e7e1f424f24ea2510a7f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32613">issues.joomla.org/tracker/joomla-cms/32613</a>.</sub>

**Comment by joomdonation** [comment]
Hmm. Change to Select Menu Item is not right to me because it indicates that we need to select a menu item for these parameters?

# For Login Form menu option
- **Menu Item Login Redirect** should not be **Home**.  If we do not choose a menu item there, users will be redirected to User Profile page. So maybe changes it to **User Profile** will be better
- **Menu Item Logout Redirect** set to **Home** is expected because if we do not choose a menu item there, users will be redirected to Home when logount using that menu option

# For Login module

- **Login Redirection Page** and **Logout Redirection Page** maybe should default to **Current Page** because if we do not select menu item there, users will be redirected to the current page after login/logout.

**Comment by chmst** [comment]
I see this PR as a temporary qick improvement until other changes are made. 

@joomdonation 
I agree with you and see here - in #32561 (https://github.com/joomla/joomla-cms/issues/32561#issuecomment-789053320)  this is already required. 
Maybe you could make the PR?

**Comment by joomdonation** [comment]
@chmst Better if @drmenzelit could update this PR if we agree with that (I'm not confident with my English :D )

**Comment by brianteeman** [comment]
@chmst Just because they are old doesn't mean they can't be improved or changed. If necessary even the description can be put back for this field. And as I am sure you know placeholder text/hints are terrible from an accessibility perspective.

However I just took a look at the actual code and "Home" is not a placeholder text or hint. It is the option which has a value of "". 

Therefore the override for this language string has always been wrong and should never have said Home and the problem is in the XML not the language string.

Instead of 
`<option value="">JDEFAULT</option>`
the xml should have been

`<option value="">JOPTION_SELECT_MENU</option>`

So the language string override should be removed from administrator/language/en-GB/com_menus.ini  and the xml file fixed as above - problem solved without the confusion of incorrect text and without overriding an override. 

It might still need a description and I see that a description is used for the other redirect method

**Comment by drmenzelit** [comment]
I will try to improve this PR. Thank you @joomdonation and @brianteeman for the comments.

**Comment by drmenzelit** [comment]
A new try ;-) I have edited the description of the PR to adapt to the changes made on my last commit.
@brianteeman I will appreciate your help with the new descriptions... you know, English is not my native language ;-)
I think `Select menu` is the correct value for the fields, it shows that an action is needed if you want to change the standard behaviour. And the standard behaviour is explained in the description.

File: components/com_users/tmpl/login/default.xml [hunk 1]
```diff
@@ -42,14 +42,15 @@
 					name="login_redirect_menuitem"
 					type="modal_menu"
 					label="COM_USERS_FIELD_LOGIN_REDIRECTMENU_LABEL"
+					description="JFIELD_LOGIN_REDIRECT_MENU_DESC"
```
Comment (brianteeman) [inline components/com_users/tmpl/login/default.xml]: Why is this JFIELD and not a COM_USERS string. Is it used outside of this component? joomla.ini should only be for global strings imho

**Comment by brianteeman** [comment]
The texts are fine - my only question is about the language file

**Comment by drmenzelit** [comment]
`JOPTION_SELECT_MENU` was already defined in joomla.ini ...

**Comment by brianteeman** [comment]
> `JOPTION_SELECT_MENU` was already defined in joomla.ini ...

That one is correct as its global.
I am referring to JFIELD_LOGIN_REDIRECT_MENU_DESC

**Comment by drmenzelit** [comment]
Sorry, I didn't read correctly... since the labels `JFIELD_LOGIN_REDIRECT_URL_LABEL` and `JFIELD_LOGOUT_REDIRECT_URL_LABEL` were also in joomla.ini, I thought it is better to put the description also there...

**Comment by brianteeman** [comment]
It makes no sense to introduce a new field into two language files and then only use one of them. That is just creating work for translators that doesn't need to be done. 

Someone else may correct me but my recollection is that joomla.ini is for global strings and as this string is only for com_users it should live in the com_users.ini

There may have been mistakes in the past but that doesnt mean we should repeat them ;)

File: components/com_users/tmpl/login/default.xml (hunk 1 above)
Comment (drmenzelit) [reply components/com_users/tmpl/login/default.xml]: Corrected in last commit. The other language strings can be moved in another PR is wished / needed.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 5adc2ca3bde69dc537c4c15147a94f256d2a3973<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32613">issues.joomla.org/tracker/joomla-cms/32613</a>.</sub>

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 5adc2ca3bde69dc537c4c15147a94f256d2a3973<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32613">issues.joomla.org/tracker/joomla-cms/32613</a>.</sub>

**Comment by chmst** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32613">issues.joomla.org/tracker/joomla-cms/32613</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32613">issues.joomla.org/tracker/joomla-cms/32613</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**